### PR TITLE
chore(deps): upgrade formatjs packages

### PR DESCRIPTION
**Pull request by longlho**
@JakeChampion this should have the UMD bundles

### CODE REVIEW

**Comment by JakeChampion** [comment]
@longlho I went through all the Intl polyfills and added to their config.toml.dependencies array all the dependencies which are used in the polyfills but were not listed.

**Comment by longlho** [comment]
thanks @JakeChampion do they need stuff all the way from ES3-?

**Comment by JakeChampion** [comment]
> thanks @JakeChampion do they need stuff all the way from ES3-?

Yes, we support IE 8 in this project

**Comment by JakeChampion** [comment]
The `Function.prototype.bind` implementation that is within the Intl UMD files does not work because it is the UMD file is running in a strict context from having `'use strict'` at the top of the UMD file.

I'm going to fix that in polyfill-library by marking `Function.prototype.bind` as a dependency, which will make it use the polyfill-library `Function.prototype.bind` polyfill instead of the one bundled in the UMD file.

This should be fixed within the `@formatjs` project as well @longlho

**Comment by JakeChampion** [comment]
@longlho 
IE 11 currently fails the tests due to the `before` block in the tests.js file throwing an error:
>Unable to get property 'full' of undefined or null reference

This error is thrown from within the `_loop_1` function.

**Comment by JakeChampion** [comment]
@longlho would it be possible to get a version of the Intl polyfills which doesn't contain extra polyfills for things which are not part of Intl? For example the UMD bundles contain polyfills for `Reflect.apply`, `Object.keys`, `Symbol.iterator`, `Object.assign` and more features.

**Comment by longlho** [comment]
hmm `Object.assign` seems to come from TypeScript, the rest from `es-abstract`. Let me see...

**Comment by longlho** [comment]
@JakeChampion I rm'ed `es-abstract` so the list of deps reduced a bit. TypeScript still injects `Object.assign` though so nothing much I can do about that other than dist es6 (which I assume the bundler won't transpile?)

**Comment by longlho** [comment]
@JakeChampion can I get a review?

**Comment by JakeChampion** [comment]
@longlho running the test suite now 👍

**Comment by JakeChampion** [comment]
@longlho CI finished now 👍  The tests failed in IE 9, 10 and 11.

To run the test server locally to help debug this you can use this command: `node ./test/polyfills/server.js` and then visit http://bs-local.com:9876/test?includePolyfills=yes&always=no&feature=Intl.DateTimeFormat in the browser you want to debug/test in.

Below are the test results for the tests which failed:
```
Failures:
- ie/9.0:
   -> Intl.DateTimeFormat passes the feature detect
      http://bs-local.com:9876/test?includePolyfills=yes&always=no&feature=Intl.DateTimeFormat
      locale data nu must be an object
   -> Intl.DateTimeFormat formatToParts should work
      http://bs-local.com:9876/test?includePolyfills=yes&always=no&feature=Intl.DateTimeFormat
      locale data nu must be an object
   -> Intl.DateTimeFormat should fix dayperiod bug in chrome
      http://bs-local.com:9876/test?includePolyfills=yes&always=no&feature=Intl.DateTimeFormat
      locale data nu must be an object
   -> Intl.RelativeTimeFormat Intl.RelativeTimeFormat format should work for style narrow
      http://bs-local.com:9876/test?includePolyfills=yes&always=no&feature=Intl.RelativeTimeFormat
      Unable to get value of the property 'indexOf': object is null or undefined
   -> Intl.RelativeTimeFormat Intl.RelativeTimeFormat formatToParts should work for numeric auto
      http://bs-local.com:9876/test?includePolyfills=yes&always=no&feature=Intl.RelativeTimeFormat
      Unable to get value of the property 'indexOf': object is null or undefined
- ie/10.0:
   -> Intl.DateTimeFormat passes the feature detect
      http://bs-local.com:9876/test?includePolyfills=yes&always=no&feature=Intl.DateTimeFormat
      locale data nu must be an object
   -> Intl.DateTimeFormat formatToParts should work
      http://bs-local.com:9876/test?includePolyfills=yes&always=no&feature=Intl.DateTimeFormat
      locale data nu must be an object
   -> Intl.DateTimeFormat should fix dayperiod bug in chrome
      http://bs-local.com:9876/test?includePolyfills=yes&always=no&feature=Intl.DateTimeFormat
      locale data nu must be an object
   -> Intl.RelativeTimeFormat Intl.RelativeTimeFormat format should work for style narrow
      http://bs-local.com:9876/test?includePolyfills=yes&always=no&feature=Intl.RelativeTimeFormat
      Unable to get property 'indexOf' of undefined or null reference
   -> Intl.RelativeTimeFormat Intl.RelativeTimeFormat formatToParts should work for numeric auto
      http://bs-local.com:9876/test?includePolyfills=yes&always=no&feature=Intl.RelativeTimeFormat
      Unable to get property 'indexOf' of undefined or null reference
- ie/11.0:
   -> Intl.DateTimeFormat passes the feature detect
      http://bs-local.com:9876/test?includePolyfills=yes&always=no&feature=Intl.DateTimeFormat
      locale data nu must be an object
   -> Intl.DateTimeFormat formatToParts should work
      http://bs-local.com:9876/test?includePolyfills=yes&always=no&feature=Intl.DateTimeFormat
      locale data nu must be an object
   -> Intl.DateTimeFormat should fix dayperiod bug in chrome
      http://bs-local.com:9876/test?includePolyfills=yes&always=no&feature=Intl.DateTimeFormat
      locale data nu must be an object
   -> Intl.RelativeTimeFormat Intl.RelativeTimeFormat format should work for style narrow
      http://bs-local.com:9876/test?includePolyfills=yes&always=no&feature=Intl.RelativeTimeFormat
      Unable to get property 'indexOf' of undefined or null reference
   -> Intl.RelativeTimeFormat Intl.RelativeTimeFormat formatToParts should work for numeric auto
      http://bs-local.com:9876/test?includePolyfills=yes&always=no&feature=Intl.RelativeTimeFormat
      Unable to get property 'indexOf' of undefined or null reference
```

**Comment by longlho** [comment]
@JakeChampion Updated CLDR was why the tests failed. I'm wondering if there's a better way to manage this other than copy-pasta?

**Comment by longlho** [comment]
@JakeChampion gentle ping?

**Comment by JakeChampion** [comment]
> @JakeChampion Updated CLDR was why the tests failed. I'm wondering if there's a better way to manage this other than copy-pasta?

Unfortunately not as of yet, sorry 😔 

> @JakeChampion gentle ping?

Yep, I've been not working for the last 4 days, looking at it now 👍

**Review by JakeChampion — APPROVED** [review]
approved, waiting on tests to pass on ci before merging

**Comment by JakeChampion** [comment]
@longlho The tests have finished running across the browsers and it currently fails on one test on ios 14:
```
Failures:
 - ios/14:
    -> Intl.RelativeTimeFormat Intl.RelativeTimeFormat format should work for style narrow
       http://bs-local.com:9876/test?includePolyfills=yes&always=no&feature=Intl.RelativeTimeFormat
Error: Failures detected
    at /home/runner/work/polyfill-library/polyfill-library/test/polyfills/remotetest.js:318:13
       in 3 qtrs. == in 3 qtr
```

**Comment by longlho** [comment]
kk I fixed that. It's because ios14 ships w/ an older version of CLDR data

File: polyfills/Intl/RelativeTimeFormat/tests.js [hunk 1]
```diff
@@ -2859,9 +2859,9 @@ describe("Intl.RelativeTimeFormat", function () {
     proclaim.equal(
       new Intl.RelativeTimeFormat("en", { style: "narrow" }).format(
         3,
-        "quarter"
+        "month"
       ),
-      "in 3 qtr"
+      "in 3 mo."
```
Comment (JakeChampion) [inline polyfills/Intl/RelativeTimeFormat/tests.js:2864]: @longlho should this be:
```suggestion
      "in 3 mo"
```

The current version fails in IE 9, 10 and 11.

```
Failures:
 - ie/10.0:
    -> Intl.RelativeTimeFormat Intl.RelativeTimeFormat format should work for style narrow
       http://bs-local.com:9876/test?includePolyfills=yes&always=no&feature=Intl.RelativeTimeFormat
       in 3 mo == in 3 mo.
 - ie/11.0:
    -> Intl.RelativeTimeFormat Intl.RelativeTimeFormat format should work for style narrow
Error: Failures detected
    at /home/runner/work/polyfill-library/polyfill-library/test/polyfills/remotetest.js:318:13
       http://bs-local.com:9876/test?includePolyfills=yes&always=no&feature=Intl.RelativeTimeFormat
       in 3 mo == in 3 mo.
 - ie/9.0:
    -> Intl.RelativeTimeFormat Intl.RelativeTimeFormat format should work for style narrow
       http://bs-local.com:9876/test?includePolyfills=yes&always=no&feature=Intl.RelativeTimeFormat
       in 3 mo == in 3 mo.
```

Comment (longlho) [reply polyfills/Intl/RelativeTimeFormat/tests.js]: I pushed a fix. This is bc the CLDR data shipped with the polyfill is newer than Chrome's version (in the new version all `.` were removed so `mo.` -> `mo`, `qtr.` -> `qtr`). IE got the polyfill so it outputs `mo` but Chrome doesn't so it outputs `mo.`

**Comment by JakeChampion** [comment]
@longlho running on CI once again 👍

**Comment by longlho** [comment]
let me bump 1 more version just released a patch for `Intl.DateTimeFormat`

**Pull request merged**